File: gitdot-api-ts/src/endpoint/migration/github/create-github-installation.ts
import { z } from "zod";
import { GitHubInstallationResource } from "../../../resource";
import type { Endpoint } from "../../endpoint";

export const CreateGitHubInstallationRequest = z.object({});
export type CreateGitHubInstallationRequest = z.infer<
  typeof CreateGitHubInstallationRequest
>;

export const CreateGitHubInstallationResponse = GitHubInstallationResource;
export type CreateGitHubInstallationResponse = z.infer<
  typeof CreateGitHubInstallationResponse
>;

export const CreateGitHubInstallation = {
  path: "/migration/github/{installation_id}",
  method: "POST",
  request: CreateGitHubInstallationRequest,
  response: CreateGitHubInstallationResponse,
} as const satisfies Endpoint;
export type CreateGitHubInstallation = typeof CreateGitHubInstallation;

File: gitdot-api/src/endpoint/migration/github/create_github_installation.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::migration::GitHubInstallationResource};

pub struct CreateGitHubInstallation;

impl Endpoint for CreateGitHubInstallation {
    const PATH: &'static str = "/migration/github/{installation_id}";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateGitHubInstallationRequest;
    type Response = CreateGitHubInstallationResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct CreateGitHubInstallationRequest {
    pub state: String,
    pub code: String,
}

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct CreateGitHubInstallationResponse {
    pub installation: GitHubInstallationResource,
    pub action: String,
}
